### PR TITLE
Improve incremental app test fixture

### PR DESCRIPTION
**Pull request by lacasseio**
### Context
Improvement to test fixture for incremental test coverage to come.

### Gradle Core Team Checklist
- [x] Verify design and implementation 
- [x] Verify test coverage and [CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Check&tab=projectOverview&branch_Gradle_Check=dl%2Fincremental-test-fixture)
- [x] Verify documentation including proper use of `@since` and `@Incubating` annotations for all public APIs
- [x] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalElement.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    }
+
+    /**
+     * Returns a transform that rename the before element to {@code renamed-} followed by the original name.
+     */
+    protected Transform rename(SourceFileElement beforeElement) {
+        assert beforeElement.getFiles().size() == 1;
+        final String sourceSetName = beforeElement.getSourceSetName();
+        final SourceFile beforeFile = beforeElement.getSourceFile();
+        final SourceFile afterFile = new SourceFile(beforeFile.getPath(), "renamed-" + beforeFile.getName(), beforeFile.getContent());
```
Comment (ghale) [inline subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalElement.java]: I wonder if this method should take a `String` argument that is the renamed file name.  You might have an overload that only takes the `SourceFileElement` and defaults to "renamed-xxx".  Not sure if there might be a scenario where the renamed file name is important or not.  Or you could just leave this as is - easy enough to add that functionality if it's ever required - your call.

File: subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalSwiftModifyExpectedOutputApp.groovy [hunk 2]
```diff
@@ -19,23 +19,21 @@ package org.gradle.nativeplatform.fixtures.app
...
-    def alternateApp = new SwiftAlternateApp()
+class IncrementalSwiftModifyExpectedOutputApp extends IncrementalSwiftElement {
+    private final greeter = new SwiftGreeter()
+    private final sum = new SwiftSum()
+    private final main = new SwiftMain(greeter, sum)
+    private final alternateMain = new SwiftAlternateMain(greeter)
 
-    IncrementalSwiftApp() {
+    IncrementalSwiftModifyExpectedOutputApp() {
         // Verify some assumptions that the tests make
```
Comment (ghale) [inline subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalSwiftModifyExpectedOutputApp.groovy]: Is this actually true?  Just glancing at the tests, this assumption did not jump out at me.

File: subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalSwiftModifyExpectedOutputAppWithLib.groovy [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+/**
+ * A Swift app with library that contains a changed source file.
+ */
+class IncrementalSwiftModifyExpectedOutputAppWithLib {
+    final lib = new IncrementalSwiftLib()
+    final main = new IncrementalSwiftAppWithDep(lib)
+
+    IncrementalSwiftModifyExpectedOutputAppWithLib() {
+        assert lib.original.files.size() > 1
```
Comment (ghale) [inline subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalSwiftModifyExpectedOutputAppWithLib.groovy]: Again, is this actually required?

File: subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalElement.java [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+        void populateBefore(List<SourceFile> beforeSourceFiles);
+
+        void populateAfter(List<SourceFile> afterSourceFiles);
+    }
+
+    /**
+     * Returns a transform that keep the before element intact.
+     */
+    protected Transform identity(final SourceFileElement element) {
```
[the remark this answers is not in the document]
Comment (lacasseio) [reply subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalElement.java]: Changed to `preserve`.

File: subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalElement.java (hunk 1 above)
Comment (lacasseio) [reply subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalElement.java]: I added an overload to choose the rename prefix.

**Pull request merged**